src/libfsm/internal.h

```diff
@@ -43,14 +43,20 @@ struct fsm_edge {
 
 struct fsm_state {
 	unsigned int end:1;
+	unsigned int reachable:1;
 
 	struct edge_set *edges; /* containing `struct fsm_edge *` */
 
 	void *opaque;
 
-	/* temporary relation between one FSM and another;
-	 * meaningful within one particular transformation only */
-	struct fsm_state *equiv;
+	/* these are only valid within one particular transformation.
+	 * expected to be NULL at start and set back to NULL after. */
+	union {
+		/* temporary relation between one FSM and another */
+		struct fsm_state *equiv;
+		/* tracks which states have been visited in walk2 */
+		struct fsm_state *visited;
+	} tmp;
```

Comment:
I'm glad to see this turned into a union; my prototype epsilon closure code has been abusing casts to stash other struct pointers in the `equiv` field, and this will let me clean that up fairly easily.